libexec/python/tests/test_core.py

```diff
@@ -149,18 +149,18 @@ def test_parse_image_uri(self):
         self.assertTrue(digest['namespace'] == 'mix')
 
         print("Case 8: Custom uri should use it.")
-        image_name = "catdog://meow/mix/%s:%s" % (self.repo_name,
-                                                  self.tag)
+        image_name = "catdog://meow/mix/tenders/%s:%s" % (self.repo_name,
+                                                          self.tag)
         digest = parse_image_uri(image_name, uri="catdog://")
         self.assertTrue(digest['registry'] == 'meow')
-        self.assertTrue(digest['namespace'] == 'mix')
+        self.assertTrue(digest['namespace'] == 'mix/tenders')
 
         print("Case 9: Digest version should be parsed")
-        image_name = ("catdog://meow/mix/%s:%s@sha:256xxxxxxxxxxxxxxx"
+        image_name = ("catdog://meow/mix/original/choice/%s:%s@sha:256xxxxxxxxxxxxxxx"  # noqa
                       % (self.repo_name, self.tag))
         digest = parse_image_uri(image_name, uri="catdog://")
         self.assertTrue(digest['registry'] == 'meow')
-        self.assertTrue(digest['namespace'] == 'mix')
+        self.assertTrue(digest['namespace'] == 'mix/original/choice')
```

Comment:
I don't understand this uri - namespace usually refers to the thing that could be thought of like a collection (eg, "library" or for singularity hub a Github repository name. In this case you are saying the namespace has slashes in it? I would have expected the registry to encompass the first bit, and then the namespace to follow. For example:

```
catdog://meow/mix/original/choice/ubuntu:latest@sha:256xxxxxxxxxxxxxxx
[   uri   ][            registry                 ][ namespace]:[ tag ]@[       digest           ]
```